kratos/containers/pointer_vector_set.h

```diff
@@ -726,81 +871,112 @@ class PointerVectorSet final
+         * @return True if the extracted key from `a` is less than the extracted key from `b`, false otherwise.
+         */
         bool operator()(TPointerType a, TPointerType b) const
         {
             return TCompareType()(TGetKeyType()(*a), TGetKeyType()(*b));
         }
     };
 
+    /**
+    * @class EqualKeyTo
+    * @brief A class providing equality comparison operators for keys in a custom map.
+    * @details This class defines equality comparison operators for keys in a custom map. It allows you to check if a key is equal to the specified key using the specified equality function and key extraction function.
+    */
     class EqualKeyTo
     {
-        key_type mKey;
+        key_type mKey;  /// The key to compare against.
     public:
+        /**
+        * @brief Default constructor.
+        * @details Initializes the `EqualKeyTo` object with a default-constructed key.
+        */
         EqualKeyTo() : mKey() {}
+
+        /**
+        * @brief Explicit constructor with a specified key.
+        * @details Initializes the `EqualKeyTo` object with the specified key.
+        * @param Key The key of type `key_type` to compare against.
+        */
         explicit EqualKeyTo(key_type Key) : mKey(Key) {}
+
+        /**
+        * @brief Compare a pointer to an object with the stored key.
+        * @details This function checks if the key stored in this `EqualKeyTo` object is equal to the extracted key from the pointer to an object of type `TPointerType`.
+        * @param a The pointer to an object of type `TPointerType`.
+        * @return True if the stored key is equal to the extracted key from `a`, false otherwise.
+        */
         bool operator()(TPointerType a) const
         {
             return TEqualType()(mKey, TGetKeyType()(*a));
         }
+
+        /**
+        * @brief Compare two pointers to objects with each other.
+        * @details This function checks if the extracted keys from two pointers to objects of type `TPointerType` are equal using the specified equality function and key extraction function.
+        * @param a The pointer to the first object of type `TPointerType`.
+        * @param b The pointer to the second object of type `TPointerType`.
+        * @return True if the extracted key from `a` is equal to the extracted key from `b`, false otherwise.
+        */
         bool operator()(TPointerType a, TPointerType b) const
         {
             return TEqualType()(TGetKeyType()(*a), TGetKeyType()(*b));
         }
     };
 
+    ///@}
     ///@name Static Member Variables
     ///@{
 
     ///@}
     ///@name Member Variables
     ///@{
 
-    TContainerType mData;
-    size_type  mSortedPartSize;
-    size_type mMaxBufferSize;
+    TContainerType mData;       /// The data container holding the elements.
+    size_type mSortedPartSize;  /// The size of the sorted portion of the data.
+    size_type mMaxBufferSize;   /// The maximum buffer size for data storage.
```

Comment:
```suggestion
    /// The data container holding the elements.
    TContainerType mData;
    
    /// The size of the sorted portion of the data.
    size_type mSortedPartSize;
    
    /// The maximum buffer size for data storage.
    size_type mMaxBufferSize;
```

Docstrings need to precede the item they belong to, otherwise `doxygen` won't be able to pick them up (or worse, it'll assume the docstring belongs to the next item).